Zend/zend_builtin_functions.c

```diff
@@ -1381,7 +1381,7 @@ static inline void get_declared_class_impl(INTERNAL_FUNCTION_PARAMETERS, int fla
 	ZEND_HASH_FILL_PACKED(Z_ARRVAL_P(return_value)) {
 		ZEND_HASH_MAP_FOREACH_STR_KEY_VAL(EG(class_table), key, zv) {
 			ce = Z_PTR_P(zv);
-			if ((ce->ce_flags & (ZEND_ACC_LINKED|ZEND_ACC_INTERFACE|ZEND_ACC_TRAIT)) == flags
+			if ((ce->ce_flags & (ZEND_ACC_LINKED|ZEND_ACC_INTERFACE|ZEND_ACC_TRAIT|ZEND_ACC_ENUM)) == flags
```

Comment:
The classes change is untested, i.e. that enums are no longer included in get_declared_classes(). That should also be mentioned in UPGRADING.

Reply:
You are right it was untested, but I also wasn't intending to exclude Enums from `get_declared_classes`. Could you help me shedding me a light how can I get Enums included for `get_declared_classes` from this bitmask check?

Reply:
Pass the ZEND_ACC_ENUM flag in get_declared_classes().

Reply:
Oh, actually that won't work. You'd need an optional flags param. That will require some more changes.

Reply:
```diff
diff --git a/Zend/zend_builtin_functions.c b/Zend/zend_builtin_functions.c
index 5759ff45a1..0bb39528cc 100644
--- a/Zend/zend_builtin_functions.c
+++ b/Zend/zend_builtin_functions.c
@@ -1381,7 +1381,7 @@ static inline void get_declared_class_impl(INTERNAL_FUNCTION_PARAMETERS, int fla
 	ZEND_HASH_FILL_PACKED(Z_ARRVAL_P(return_value)) {
 		ZEND_HASH_MAP_FOREACH_STR_KEY_VAL(EG(class_table), key, zv) {
 			ce = Z_PTR_P(zv);
-			if ((ce->ce_flags & (ZEND_ACC_LINKED|ZEND_ACC_INTERFACE|ZEND_ACC_TRAIT|ZEND_ACC_ENUM)) == flags
+			if ((ce->ce_flags & (flags|ZEND_ACC_LINKED|ZEND_ACC_INTERFACE|ZEND_ACC_TRAIT)) == flags
 			 && key
 			 && ZSTR_VAL(key)[0] != 0) {
 				ZEND_HASH_FILL_GROW();

```

That would do.

Reply:
Brilliant! Thank you so much 🙏. I included this, along with a few more test assertions.